### DISCUSSION TITLE
How to access process.env?

### DISCUSSION THREAD

**Discussion by agriffis** [babashka/nbb#276] [Q&A] [answered]
I'm just getting started with nbb. I'm struggling to access process.env. Could someone point me in the right direction?

**Comment by borkdude** [comment, marked as answer]
In CLJS you write `js/process.env` so `js/something` to get access to the global `something`.

**Reply by borkdude** [reply]
The more official syntax is `(.-env js/process)` but `js/process.env` also works

**Reply by agriffis** [reply]
Thanks! That seems to return something that I'm not sure how to access, and `js->clj` doesn't seem to have any effect.

```
user=> js/process.env
#object[Object [object Object]]
user=> (js->clj js/process.env)
#object[Object [object Object]]
```

**Reply by borkdude** [reply]
@agriffis You can probably best call `(js/console.log js/process.env)` to print the JS object using `console.log`

**Reply by agriffis** [reply]
Well, I really want to access it as a Clojure map, not just print it. There is some suggestion at https://stackoverflow.com/q/32467299/347386 that the problem is that `process.env` isn't a plain object, and so `js->clj` doesn't work. But the various answers there reference `goog.isObject` and `goog/typeOf` that doen't seem to be available in nbb.

**Reply by agriffis** [reply]
Got something here:

```
(into {} (map vector (goog.object/getKeys js/process.env) (goog.object/getValues js/process.env))
```

**Reply by borkdude** [reply]
Ah yes, I also was trying to figure it out:

``` clojure
(into {} (map js->clj (js/Object.entries js/process.env)))
```

**Reply by agriffis** [reply]
I guess I also need to set things in `process.env` but I can probably figure it out. Thanks for the help!
